test/unit/three_scale/api/responder_test.rb

```diff
@@ -18,7 +18,7 @@ module Representer
 
   test '#to_format PATCH' do
     @controller.formats = [:json]
-    @controller.request = ActionDispatch::Request.new('REQUEST_METHOD' => 'PATCH')
+    @controller.request = ActionDispatch::Request.new('REQUEST_METHOD' => 'PATCH', 'rack.input' => StringIO.new)
```

Comment:
This one made me scratch my head quite a bit...

The error was:
```

Error:
ThreeScale::Api::ResponderTest#test_#to_format_PATCH:
RuntimeError: Missing rack.input
    test/unit/three_scale/api/responder_test.rb:25:in `block in <class:ResponderTest>'
```

Debugging showed that it was caused by https://github.com/rack/rack/blob/v2.2.8.1/lib/rack/request.rb#L441-L442

And the reason why this called in the first place was this addition:

![Screenshot from 2024-03-14 13-40-27](https://github.com/3scale/porta/assets/1270328/5bbd4178-0be5-44c8-8b1f-448e7c98e051)

added here: https://github.com/rails/rails/commit/5745a3c0928ee5604ce80af19348efb42189f1d6

By the way, `rack.input` was later made optional in rack: https://github.com/rack/rack/pull/2018

But this version is not released yet.

I think this really happens just because we are creating this fake request, and in real-life scenario this would not cause issues, so I think adding `rack.input` to the `env` in Request initializer should do the trick. But I don't understand this test enough to be sure :grimacing:

Reply:
maybe put a comment that this can be removed after we have a rack version with https://github.com/rack/rack/pull/2018